Edits/New Term Entries - Java: Arrays and ArrayList (#608)
Codecademy/docs: content/java/concepts/arrays/arrays.md

---
Title: 'Arrays'

> Description: 'Arrays are used for storing elements of the same data type in a fixed collection, ordered by insertion.'

Reviewer: Can we say 
"fixed-sized"?
```suggestion
Description: 'Arrays are used for storing elements of the same data type in a fixed-sized collection, ordered by insertion.'
```

Author: Absolutely! 👍

Subjects:
  - 'Code Foundations'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Data Types'
  - 'Data Structures'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> In Java, an array is used for storing elements of the same data type in a fixed collection, ordered by insertion.

Reviewer: Can we clarify what fixed means? It can be different from my suggestion.
```suggestion
In Java, an array is used for storing elements of the same data type in a fixed-sized collection, ordered by insertion. Being fixed in size means that unlike with [ArrayList](https://www.codecademy.com/resources/docs/java/array-list), elements cannot be removed or added after the array has been created. They can only be modified.
```

## Syntax

Array variables are defined with a type followed by square brackets `[]`. Next, the actual elements are comma-separated and surrounded by curly brackets `{}`:

```pseudo
dataType[] varName = {elemOne, elemTwo, elemThree, elemN};
```

Alternatively, an array can be defined with the `new` keyword, the `arrayLength` integer defined, and each of the elements defined afterward:

```pseudo
dataType[] varName = dataType[arrayLength];

varName[0] = valueOne;
varName[1] = valueTwo;
varName[n] = valueN;
```

It is important the referenced index is valid. Otherwise, an `ArrayIndexOutOfBoundsException` will be thrown.

## Example

In the example below, an array can be created in the following ways:

```java
int[] semesterOneMarks = {10, 20, 30, 40, 50};

String[] semesterOneCourses = new String[5];

semesterOneCourses[0] = "Algebra";
semesterOneCourses[1] = "World History";
semesterOneCourses[2] = "Literature";
semesterOneCourses[3] = "Programming";
semesterOneCourses[4] = "Gym";
```

> Their elements can be accessed and changed as well:

Reviewer: ```suggestion
Elements within the array can be accessed and changed as well:
```
